staging/src/k8s.io/apimachinery/pkg/watch/tools/retrywatcher.go

```diff
@@ -0,0 +1,154 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tools
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+// WatchFunc is a function that is responsible for creating a watcher starting at sinceResourceVersion.
+type WatchFunc func(sinceResourceVersion string) watch.Interface
+
+// RetryWatcherError is the type of an event that will be returned if RetryWatcher fails.
+type RetryWatcherError string
+
+// GetObjectKind implements interface.
+func (obj RetryWatcherError) GetObjectKind() schema.ObjectKind { return schema.EmptyObjectKind }
+
+// DeepCopyObject implements interface.
+func (obj RetryWatcherError) DeepCopyObject() runtime.Object { return obj }
```

Comment:
Why should this be a runtime.Object?

Reply:
it is returned as an object in event when error occurs https://github.com/tnozicka/kubernetes/blob/1c33685d5b4c2eaa14a7650dc7f79eeeb195e31c/staging/src/k8s.io/apimachinery/pkg/watch/tools/retrywatcher.go#L95

Reply:
Is `RetryWatcherError` exposed to the user of the retry watcher? Wouldn't wrapping `error` make more sense? Or even `Status`.